types/vote.go

```diff
@@ -410,33 +416,31 @@ func SignAndCheckVote(
 	v := vote.ToProto()
 	if err := privVal.SignVote(chainID, v); err != nil {
 		// Failing to sign a vote has always been a recoverable error, this function keeps it that way
-		return true, err // true = recoverable
+		return true, err
 	}
 	vote.Signature = v.Signature
+	vote.ExtensionSignature = nil
+	vote.Timestamp = v.Timestamp
 
-	isPrecommit := vote.Type == PrecommitType
-	if !isPrecommit && extensionsEnabled {
-		// Non-recoverable because the caller passed parameters that don't make sense
-		return false, fmt.Errorf("only Precommit votes may have extensions enabled; vote type: %d", vote.Type)
-	}
-
-	isNil := vote.BlockID.IsNil()
-	extSignature := (len(v.ExtensionSignature) > 0)
-	if extSignature == (!isPrecommit || isNil) {
```

Comment:
previously this would error if 

* extSignature is empty
* recommit is true
* isNil is false

so `false == (!true || false), false == false`